File: src/Application.Models/Employees/EmployeeModel.cs
namespace Application.Models.Employees;

public class EmployeeModel
{
    public string Id { get; set; }

    public string FirstName { get; set; } = string.Empty;

    public string LastName { get; set; } = string.Empty;

    public string DocNumber { get; set; } = string.Empty;

    public short DocType { get; set; } = 0;

    public IEnumerable<RegisterPhoneModel> Phones { get; set; } = new List<RegisterPhoneModel>();

    public string Email { get; set; } = string.Empty;

    public string BirthDate { get; set; } = string.Empty;

    public string? ManagerId { get; set; }

    public string? Role { get; set; }
}

File: angular-employee-management/src/app/models/employee-model.ts
import { RegisterPhoneModel } from "./register-phone-model";
import { Role } from "./role.enum";

export class EmployeeModel {
    id: string = '';
    email: string = '';
    firstName: string = '';
    lastName: string = '';
    docNumber: string = '';
    docType: number = 0;
    phones: RegisterPhoneModel[] = [];
    birthDate: string = '';
    managerId?: string;
    password: string = '';
    role: Role = Role.Employee;
  
    constructor(init?: Partial<EmployeeModel>) {
      Object.assign(this, init);
    }
}
